### PR TITLE
feat: allow any type of filter to be used

### PR DESCRIPTION
**Pull request by fdlk**
The filter list can contain any type of filter component, it need not be specified in this library.
Use `<component :is="string-filter">` to set it.

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] Conventional commits (squash if needed)
- [x] No warnings during install
- [x] Updated javascript typing

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -20,6 +20,8 @@
 <script lang="ts">
 import Vue from 'vue'
 import FilterContainer from './components/FilterContainer.vue'
+import CheckboxFilter from './components/CheckboxFilter.vue'
+import StringFilter from './components/StringFilter.vue'
```
Comment (MaxPostema) [inline src/App.vue]: Suggest to collapse filter imports to single line
`import { FilterContainer, StringFilter, CheckboxFilter } from './components'`

File: src/App.vue [hunk 2]
```diff
@@ -38,54 +40,54 @@ export default Vue.extend({
         description: 'search by name',
         initiallyCollapsed: true,
         placeholder: 'test',
-        type: 'string'
+        type: StringFilter
```
Comment (connoratrug) [inline src/App.vue]: This setup makes hard to store the filter setup, this seems like a big problem to me geven the features needed, suggest to discuss this.

Comment (fdlk) [reply src/App.vue]: Perhaps if they're registered, you can also use the name?

Comment (fdlk) [reply src/App.vue]: I've updated the App.vue to show that yes, this actually works

**Pull request merged**